apps/sync/js/adapters/bookmarks.js

```diff
@@ -116,14 +116,14 @@ var BookmarksHelper = (() => {
     var revisionId;
     return _ensureStore().then(store => {
       revisionId = store.revisionId;
-      return store.get(id);
-    }).then(localRecord => {
-      if (localRecord) {
-        var newBookmark = mergeRecordsToDataStore(localRecord, remoteRecord);
-        return store.put(newBookmark, id, revisionId);
```

Comment:
Not functionally (after all, the tests are passing currently), but I changed it mainly because the code would be harder to maintain, because it breaks the encapsulation that `_ensureStore` creates, and to make the code less surprising to the reader.